client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,319 @@
+ *        <section id="tab-tab-1" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-1" data-w-tabs-target="panel">
+ *            tab-1
+ *        </section>
+ *        <section id="tab-tab-2" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-2" data-w-tabs-target="panel">
+ *            tab-2
+ *        </section>
+ *    </div>
+ *</div>
+ */
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['list', 'trigger', 'label', 'panel'];
+
+  static classes = ['selected'];
+
+  static values = {
+    transition: { default: 150, type: Number },
+    selected: { default: '', type: String },
+    syncURLHash: { default: false, type: Boolean },
+    animate: { default: true, type: Boolean },
+  };
+
+  declare readonly selectedClasses: string;
+
+  declare listTarget: HTMLDivElement;
+  declare triggerTargets: HTMLAnchorElement[];
+  declare labelTargets: HTMLAnchorElement[];
+  declare panelTargets: HTMLElement[];
+
+  declare syncURLHashValue: boolean;
+  declare selectedValue: string;
+  declare transitionValue: number;
+  declare animateValue: boolean;
+
+  connect() {
+    this.validate();
+
+    debounce(() => {
+      window.scrollTo({ top: 0, behavior: 'smooth' });
+    }, this.transitionValue * 2);
+
+    this.setAriaControls(this.labelTargets);
+    this.setTabLabelIndex();
+
+    const activeTab = this.labelTargets.find(
+      (button) =>
+        button.getAttribute('aria-selected') === 'true' ||
+        button.getAttribute('aria-controls') === this.selectedValue,
+    );
+
+    if (this.selectedClasses !== '' && activeTab) {
+      activeTab.setAttribute('aria-selected', 'true');
+      activeTab.removeAttribute('tabindex');
+    }
+
+    this.panelTargets.forEach((tab) => {
+      // eslint-disable-next-line no-param-reassign
+      tab.hidden = true;
+    });
+
+    // console.log(this.selectedValue ? "hello": "bye!", "selected")
+    // console.log(activeTab)
+    if (window.location.hash && !this.syncURLHashValue) {
+      this.setTabByURLHash();
+    } else if (activeTab) {
+      this.selectedValue = activeTab.getAttribute('aria-controls') as string;
+    } else {
+      this.selectFirstTab();
+    }
+
+    this.setAriaControls(this.triggerTargets);
+  }
+
+  selectedValueChanged(currentValue: string, previousValue: string) {
+    if (previousValue) {
+      this.hideTabContent(previousValue);
+    }
+
+    const tab = this.getTabLabelByHref(currentValue);
+    if (tab) {
+      tab.setAttribute('aria-selected', 'true');
+      tab.removeAttribute('tabindex');
+    }
+
+    const tabContent = this.getTabPanelByHref(currentValue);
+
+    if (tabContent) {
+      if (this.animateValue) {
+        this.animateIn(tabContent);
+      } else {
+        tabContent.hidden = false;
+      }
+
+      this.dispatch('switch', {
+        detail: { tab: tab?.getAttribute('href')?.replace('#', '') },
+        target: this.listTarget,
```

Comment:
```suggestion
```

```suggestion
        target: tab,
```

We can just dispatch this either on the tab that got switched or on the controlled element.